haystack/nodes/preprocessor/splitter.py

```diff
@@ -0,0 +1,755 @@
+
+        :param nltk_folder: If `split_by="sentence"`, specifies the path to the folder containing the NTLK `PunktSentenceTokenizer` models,
+                            if loading a model from a local path. Leave empty otherwise.
+
+        :param progress_bar: Whether to show a progress bar.
+
+        :param add_page_number: Add the number of the page a paragraph occurs in to the Document's meta
+                                field `"page"`. Page boundaries are determined by `"\f"' character which is added
+                                in between pages by `PDFToTextConverter`, `TikaConverter`, `ParsrConverter` and
+                                `AzureConverter`.
+        """
+        super().__init__()
+        self._validate_split_params(
+            split_by=split_by,
+            split_regex=split_regex,
+            split_length=split_length,
+            split_overlap=split_overlap,
+            max_chars=max_chars,
+            max_tokens=max_tokens,
+        )
+        try:
+            nltk.data.find("tokenizers/punkt")
+        except LookupError:
+            nltk.download("punkt")
+
+        self.split_by = split_by
+        self.split_regex = split_regex
+        self.max_chars = max_chars
+        self.max_tokens = max_tokens
+        self.progress_bar = progress_bar
+        self.add_page_number = add_page_number
+
+        self.merger = DocumentMerger(
+            separator="",
+            window_size=split_length,
+            window_overlap=split_overlap,
+            max_tokens=max_tokens,
+            retain_page_number=True,
+            realign_headlines=True,
+        )
+
+        self._nltk_language = nltk_language
+        self._nltk_folder = Path(nltk_folder) if nltk_folder else None
+        self.sentence_tokenizer = load_sentence_tokenizer(
+            language=self._nltk_language, tokenizer_model_folder=self._nltk_folder
+        )
+
+        self._tokenizer = None
+        if tokenizer_model or max_tokens:
+            self.tokenizer = tokenizer_model
+
+        if split_by == "token" and not self.tokenizer:
+            raise ValueError(
+                "If you set split_by='token', you must give a value to 'tokenizer_model'. "
+                "Use the same model you're using for your Reader."
+            )
+
+    def _validate_split_params(
+        self,
+        split_by: SplitBy,
+        split_regex: Optional[str],
+        split_length: int,
+        split_overlap: int,
+        max_chars: int,
+        max_tokens: Optional[int],
+    ):
+        """
+        Performs some basic validation on the parameters of the splitter.
+        """
+        if split_by not in get_args(SplitBy):
+            raise ValueError(f"split_by must be one of: {', '.join(get_args(SplitBy))}")
+
+        if not isinstance(split_length, int) or split_length < 0:
+            raise ValueError("split_length must be an integer >= 0")
+
+        if split_length == 0 and not max_tokens:
+            logger.warning(
```

Comment:
I think it would be good to also add a check here to make sure that if `max_tokens` is provided that it is larger than 0, like you do for `split_overlap`.